### DISCUSSION TITLE
Edge attributes for GatedGraphConv

### DISCUSSION THREAD

**Discussion by rubyzhou2014** [pyg-team/pytorch_geometric#2478] [Q&A]
Hi, 
I am building a model using [GatedGraphConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html?highlight=GatedGraphConv#torch_geometric.nn.conv.GatedGraphConv) operator. I want to involve edge attributes in the model training but didn't find a way. I see that it can pass edge weight into GatedGraphConv, but not edge_attr. 

Is there any API provided to support this feature? Or any suggestions for coding it manually? Thanks!

**Comment by rusty1s** [comment]
You might want to extend the `GatedGraphConv` manually for this. Usually, this is done by concatenating edge features and source node features in `message`:
```python
def message(self, x_j, edge_attr):
        return self.lin(torch.cat([x_j, edge_attr], dim=-1))
```
where `self.lin` is a `Lienar` layer that maps to `out_channels`.
